pointmatcher/DataPointsFilters/AddDescriptor.cpp

```diff
@@ -0,0 +1,60 @@
+// kate: replace-tabs off; indent-width 4; indent-mode normal
+// vim: ts=4:sw=4:noexpandtab
+
+#include "AddDescriptor.h"
+
+template <typename T>
+AddDescriptorDataPointsFilter<T>::AddDescriptorDataPointsFilter(const Parameters& params) :
+	PointMatcher<T>::DataPointsFilter("AddDescriptorDataPointsFilter",
+		AddDescriptorDataPointsFilter::availableParameters(), params),
+	descriptorName(Parametrizable::get<std::string>("descriptorName")),
+	descriptorDimension(Parametrizable::get<std::size_t>("descriptorDimension")),
+	descriptorValues(Parametrizable::getVector<T>("descriptorValues"))
+{
+    assert(descriptorDimension == descriptorValues.size());
+}
+
+// AddDescriptorDataPointsFilter
+template<typename T>
+typename PointMatcher<T>::DataPoints AddDescriptorDataPointsFilter<T>::filter(
+	const DataPoints& input)
+{
+	DataPoints output(input);
+	inPlaceFilter(output);
+	return output;
+}
+
+// In-place filter
+template<typename T>
+void AddDescriptorDataPointsFilter<T>::inPlaceFilter(
+        DataPoints& cloud)
+{
+    Matrix matrix = PM::Matrix::Ones(descriptorDimension, cloud.getNbPoints());
+    for(std::size_t i = 0; i < descriptorDimension; ++i)
+    {
+        matrix.row(i) *= descriptorValues[i];
```

Comment:
why is it a multiplicating instead of overwriting

Reply:
The Matrix `matrix` is initialized as Ones and each row is multiplicated by the corresponding descriptor value. Another option would be to assign `descriptorValues` to each column on `matrix`. Idk what's more optimal